src/Root.ts

```diff
@@ -486,9 +486,14 @@ export class RootAPI implements Types.RootAPI {
                 }
             }
 
-            if (!mover && curMover && (!modalizer || curModalizer)) {
+            if (
+                !mover &&
+                curMover &&
+                (!modalizer || curModalizer) &&
+                (!curGroupper || curElement !== element)
+            ) {
                 mover = curMover;
-                isGroupperFirst = !!groupper;
+                isGroupperFirst = !!groupper && groupper !== curGroupper;
```

Comment:
```suggestion
                isGroupperFirst = groupper !== curGroupper;
```
Wouldn't this be enough?

Reply:
No, because `groupper` can be undefined.